sympy/integrals/prde.py

```diff
@@ -441,6 +441,52 @@ def prde_no_cancel_b_small(b, Q, n, DE):
     return f + H, A.col_join(B).col_join(C)
 
 
+def prde_cancel_liouvillian(b, Q, n, DE):
+    """
+    Pg, 237.
+    Examples of this are:
+    Elementary
+    ==========
+    log(x) - log(x - 1)
+
+    Non-elementary
+    ==============
+    log(x + 1)*log(x), log(x + 1)/log(x)
+
+    """
+    m = len(Q)
+    f, A, Qy = [None]*(n + 1), [None]*(n + 1), [None]*(n + 1)
+    F = [None]*(n + 1)
+
+    # Why use DecrementLevel? Below line answers that:
+    # Assuming that we can solve such problems over 'k' (not k[t])
+    if DE.case == 'primitive':
+        with DecrementLevel(DE):
+            ba, bd = frac_in(b, DE.t, field=True)
+
+    for i in range(n, -1, -1):
+        with DecrementLevel(DE):
+            # there is no-need to recheck this
+            if DE.case == 'exp':
+                ba, bd = frac_in(b + i*derivation(DE.t, DE)/DE.t,
+                                DE.t, field=True)
+            Qyi = [frac_in(q.nth(i), DE.t, field=True) for q in Q]
+            Qy[i] = Qyi
+            fn, A[i] = param_rischDE(ba, bd, Qyi, DE)
+        f[i] = [Poly(fa.as_expr()/fd.as_expr(), DE.t, field=True)
+                for fa, fd in fn]
+
+        ri = len(fn)
+        Fi = F[i] = [None]*ri
+
+        for j in range(ri):
+            hji = f[i][j]*DE.t**i
+            Fi[j] = derivation(hji, DE) - b*hji
+        # in the next loop instead of Qy it has
+        # to be Qy + Fi taking its place, like below?
+        Qy += Fi
```

Comment:
If `Q` is the original list and `Qy` is the extended list then this should be `Qy`. But it seems that the extended list is also denoted by `Qyi` above. Are they supposed to be different?